jenkins/databricks/test.sh

```diff
@@ -36,11 +61,34 @@ sudo "$(which pip)" install pytest sre_yield requests pandas pyarrow findspark p
 export SPARK_HOME=/databricks/spark
 # change to not point at databricks confs so we don't conflict with their settings
 export SPARK_CONF_DIR=$PWD
-export PYTHONPATH=$SPARK_HOME/python:$SPARK_HOME/python/pyspark/:$SPARK_HOME/python/lib/py4j-0.10.9-src.zip
-if [[ $BASE_SPARK_VERSION == "3.2.1" ]]
+# Get Python version from environment after lowercase and remove minor.
+# Something like python3.8 for DB10.4 and python3.9 for DB11.3
+sw_versions[PYTHON]=$(python --version | sed 's/[A-Z]/\L&/g;s/ //g' | cut -d . -f 1,2)
+# Set Iceberg related versions. See https://iceberg.apache.org/multi-engine-support/#apache-spark
+case "$BASE_SPARK_VERSION" in
+    "3.3.0")
+        sw_versions[ICEBERG]=${ICEBERG_VERSION:-'0.14.1'}
+        ;;
+    "3.2.1")
+        sw_versions[ICEBERG]=${ICEBERG_VERSION:-'0.13.2'}
+        ;;
+    "3.1.2")
+        sw_versions[ICEBERG]=${ICEBERG_VERSION:-'0.13.2'}
+        ;;
+    *) echo "Unexpected Spark version: $BASE_SPARK_VERSION"; exit 1;;
+esac
+# Set the iceberg_spark to something like 3.3 for DB11.3, 3.2 for DB10.4
+sw_versions[ICEBERG_SPARK]=$(echo $BASE_SPARK_VERSION | cut -d. -f1,2)
+# Set ICEBERG scala version.
+sw_versions[ICEBERG_SCALA]=${ICEBERG_SCALA_VERSION:-'2.12'}
+# Get the correct py4j file.
+PY4J_FILE=$(find $SPARK_HOME/python/lib -type f -iname "py4j*.zip")
+export PYTHONPATH=$SPARK_HOME/python:$SPARK_HOME/python/pyspark/:$PY4J_FILE
+if [[ $BASE_SPARK_VERSION == "3.2.1" ]] || [[ $BASE_SPARK_VERSION == "3.3.0" ]]
```

Comment:
why is it conditional on the Spark version as a proxy for db runtime version? 
 
would this not work on 9.1? 
```
export PYTHONPATH=/databricks/python3/lib/${sw_versions[PYTHON]}/site-packages:$PYTHONPATH
```

Reply:
Interesting..This condition was added to support 10.4 so I didn't want to mess out with it.
According to DB documentation https://docs.databricks.com/archive/legacy/koalas.html

> Note
> Koalas is deprecated. If you try using Koalas on clusters that run [Databricks Runtime 10.0 (Unsupported)](https://docs.databricks.com/release-notes/runtime/10.0.html) and above, an informational message displays, recommending that you use [Pandas API on Spark](https://docs.databricks.com/pandas/pandas-on-spark.html) instead.

This means that we should not expect Koala on 10.4+. The original condition might be negated here.
I removed the condition in this PR.

@jlowe any thoughts?

Reply:
I don't know the full context behind this, other than we were seeing failures during the test collection phase on startup due to "bad" pandas ordering on the Python system path due to Koalas conflicting with the real Pandas modules such that one could no longer successfully import pandas.

You'll know whether or not it worked based on whether you can get past the test collection phase for udf_test.py and udf_cudf_test.py, as they both try to import pandas.